starlette/staticfiles.py

```diff
@@ -85,6 +100,10 @@ def not_modified_response(self, stat_headers: dict) -> Response:
         if self.check_directory is not None:
             await self.check_directory_configured_correctly()
 
+        (head, tail) = os.path.split(self.path)
+        if self.serve_index_html and tail == ".":
+            self.path = os.path.join(head, "index.html")
+
         try:
             stat_result = await aio_stat(self.path)
         except FileNotFoundError:
```

Comment:
We could also add a redirect case here, eg. if `"/test"` is used and `"/test/index.html" exists, then redirect to `"/test/"`

Reply:
I think that's a great idea. I played around with it last night for a couple minutes and ran into some issues, so I'll see if I can get something working this weekend.

Reply:
Alright, so here's my issue. The `Router` class has an attribute called `redirect_slashes` which is by default set to `True`. Because of this, as far as the `_StaticFilesResponder` class is concerned the paths `http://localhost:8000/static` and `http://localhost:8000/static/` are identical, as a `302 Redirect` has already occurred by the time the request reaches this class.

Therefore, redirecting from `/static` to `/static/` ends up in an infinite loop of redirects. Unfortunately I'm just not seeing the clear path forward here, so any input would be greatly appreciated.